### PR TITLE
Update developer guide requirements

### LINKED ISSUE #21: Update the Developer Guide
Do the following:
1. Update the target user profile and value proposition to match the project direction you have selected.
2. Give use cases (textual form) for a few representative user stories that need multiple steps to complete.
3. Non-functional requirements
4. Glossary

### PR DESCRIPTION
**Pull request by Assyarul**
Fixes #21

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -281,28 +320,105 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+* 3a. The given date is invalid
+  
+    * 3a1. FriendDex shows an error message, and the accepted date format(s)
+      
+      Use case ends
 
-### Glossary
+**Use case: Create Friend Groups**
 
-* **Mainstream OS**: Windows, Linux, Unix, OS-X
```
Comment (chesterhow) [inline docs/DeveloperGuide.md:302]: I think we should keep `Mainstream OS` in the glossary as mentioned in our NFRs

**Review by chesterhow — CHANGES_REQUESTED** [review]

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (Assyarul) [reply docs/DeveloperGuide.md:302]: Sounds good. Added the glossary back in.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -236,43 +236,82 @@ _{Explain here how the data archiving feature will be implemented}_
...
-**Value proposition**: manage contacts faster than a typical mouse/GUI driven app
+* is a young programmer studying in NUS Computer Science
+* prefers typing to mouse interactions  
+* has difficulty maintaining personal relationships
+* cannot recall previous conversations with others
+* is a goal-driven individual and seeks new challenges.
 
+**Value proposition**:
+* makes relationship management easier for the user
+    * keeps track of personal relationship with users' friends
```
Comment (ivantjh) [inline docs/DeveloperGuide.md:248]: Formatting!

**Review by ivantjh — CHANGES_REQUESTED** [review]

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (Assyarul) [reply docs/DeveloperGuide.md:248]: Fixed

**Review by chesterhow — APPROVED** [review]
LGTM

**Review by ivantjh — APPROVED** [review]
lgtm!

**Pull request merged**